core/src/com/softwarearchitecture/game_server/states/HostLobby.java

```diff
@@ -0,0 +1,123 @@
+package com.softwarearchitecture.game_server.states;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+
+import com.badlogic.gdx.graphics.Texture;
+import com.badlogic.gdx.graphics.g2d.BitmapFont;
+import com.badlogic.gdx.graphics.g2d.SpriteBatch;
+import com.softwarearchitecture.GameApp;
+import com.softwarearchitecture.game_server.buttons.Button;
+import com.softwarearchitecture.game_server.buttons.Factory;
+import com.softwarearchitecture.game_server.buttons.Observer;
+import com.softwarearchitecture.game_server.buttons.TypeEnum;
+import com.softwarearchitecture.game_server.buttons.GridLayout;
+import com.softwarearchitecture.math.Rectangle;
+
+public class HostLobby extends State implements Observer {
+
+    private String lobbyCode;
+    private BitmapFont font;
+
+    public HostLobby() {
+        super();
+        List<TypeEnum> buttonTypes = new ArrayList<>();
+        buttonTypes.add(TypeEnum.GAME_MENU);
+        buttons = createButtons(buttonTypes);
+
+        // placeholder background logic not implemented
+        background = TexturePack.BACKGROUND_TOR;
+
+        this.lobbyCode = generateLobbyCode(6); // Generate a 6-character code
+        this.font = new BitmapFont(); // Initialize your font (consider using a specific font file)
+    }
+
+    private static String generateLobbyCode(int length) {
+        String characters = "ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
```

Comment:
@mvbryne Nice! Btw if this is algorithm has a formal name it could be nice to add to an javadocs comment.